carmin-server/server/resources/path.py

```diff
@@ -110,9 +112,10 @@ def delete(self, user, complete_path: str = ''):
         requested_data_path = os.path.normpath(
             os.path.join(data_path, complete_path))
 
-        if (is_root_dir_for_user(requested_data_path, user)
+        if (is_root_dir_for_user(requested_data_path, user.username)
```

Comment:
I would maybe encapsulate this into a helper function named `is_safe_for_delete`.